core/src/main/java/org/jdbi/v3/core/internal/SqlScriptParser.java

```diff
@@ -90,4 +86,56 @@ public String parse(CharStream charStream) {
     public interface TokenHandler {
         void handle(Token t, StringBuilder sb);
     }
+
+    public static final class ScriptTokenHandler implements TokenHandler {
+
+        private final List<String> statements = new ArrayList<>();
+        private Token lastToken = null;
+
+        private int blockLevel = 0;
+
+        @Override
+        public void handle(Token t, StringBuilder sb) {
+            switch (t.getType()) {
+                case BLOCK_BEGIN:
```

Comment:
This could be an opportunity to use the new switch `case X -> { }` style which doesn't need `break;` statements

Reply:
... introduced in java 12 ... :-(